### PR TITLE
Enhance SystemPatchApplier for LTS version support

### PR DESCRIPTION
**Pull request by ViacheslavKlimov**
## Summary
- Relax version check (`isPatchVersionChanged` → `isVersionIncreased`) to trigger on maintenance digit increases within the same major.minor family (e.g. 4.3.0→4.3.1), not just patch digit changes
- Add LTS SQL schema patch execution from `upgrade/lts/schema_update.sql`, running before views and widget type updates
- Add placeholder `schema_update.sql` for LTS cumulative idempotent migrations

## Test plan
- [ ] `SystemPatchApplierTest` — 14 parameterized version comparison cases, 4 `isVersionChanged` tests, 3 `updateLtsSqlSchema` tests, 5 `applyPatchIfNeeded` flow tests
- [ ] Verify existing widget/view/lock tests still pass

### CODE REVIEW

File: application/src/test/java/org/thingsboard/server/system/SystemPatchApplierTest.java [hunk 1]
```diff
@@ -373,6 +377,247 @@ void whenLockIsHeldByOneThread_thenSecondThreadCannotAcquireLock() throws Except
...
+        when(installScripts.getDataDir()).thenReturn(dataDir.toString());
+
+        Path widgetTypesDir = tempDir.resolve("widget_types");
+        Files.createDirectories(widgetTypesDir);
+        when(installScripts.getWidgetTypesDir()).thenReturn(widgetTypesDir);
+
+        ReflectionTestUtils.invokeMethod(reconciler, "applyPatchIfNeeded");
+
+        // LTS SQL was executed
+        verify(jdbcTemplate).execute("SELECT 1;");
```
Comment (ViacheslavKlimov) [inline application/src/test/java/org/thingsboard/server/system/SystemPatchApplierTest.java:561]: The test name says "Applies LTS SQL **Before** Views And Widgets" but the verifications here only check that both calls happened — they don't verify the order. You could use Mockito's `InOrder` to actually assert the sequence:

```java
InOrder inOrder = inOrder(jdbcTemplate, schemaSettingsService);
inOrder.verify(jdbcTemplate).execute("SELECT 1;"); // LTS SQL first
// updateSqlViews also calls jdbcTemplate.execute, so verify that too
inOrder.verify(schemaSettingsService).updateSchemaVersion(); // version update last
```

Without ordering verification, someone could accidentally reorder the calls in `applyPatchIfNeeded` and this test would still pass.

**Review by ViacheslavKlimov — COMMENTED** [review]
## Review summary

Reviewed 3 changed files in [Enhance SystemPatchApplier for LTS version support](https://github.com/thingsboard/thingsboard/pull/15438). Left 1 comment(s) inline.

The version comparison logic in `isVersionIncreased` is correct and the test coverage is thorough — the 14 parameterized version comparison cases cover all the important edge cases (same family increases, decreases, cross-family changes). The `updateLtsSqlSchema` method handles the missing-file case gracefully and is consistent with how `updateSqlViews` works in the same class. The cumulative idempotent SQL file approach is well-documented.

---
*This review was auto-generated. Findings may contain errors — please verify before applying changes.*
